java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,343 @@
+                        smallIntVector.setSafe(rowCount, rs.getInt(i));
+                        smallIntVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.INTEGER:
+                        IntVector intVector = (IntVector)root.getVector(columnName);
+                        intVector.setSafe(rowCount, rs.getInt(i));
+                        intVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.BIGINT:
+                        BigIntVector bigIntVector = (BigIntVector)root.getVector(columnName);
+                        bigIntVector.setSafe(rowCount, rs.getInt(i));
+                        bigIntVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.NUMERIC:
+                    case Types.DECIMAL:
+                        DecimalVector decimalVector = (DecimalVector)root.getVector(columnName);
+                        decimalVector.setSafe(rowCount, rs.getBigDecimal(i));
+                        decimalVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.REAL:
+                    case Types.FLOAT:
+                        Float4Vector float4Vector = (Float4Vector)root.getVector(columnName);
+                        float4Vector.setSafe(rowCount, rs.getFloat(i));
+                        float4Vector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.DOUBLE:
+                        Float8Vector float8Vector = (Float8Vector)root.getVector(columnName);
+                        float8Vector.setSafe(rowCount, rs.getDouble(i));
+                        float8Vector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.CHAR:
+                    case Types.NCHAR:
+                    case Types.VARCHAR:
+                    case Types.NVARCHAR:
+                    case Types.LONGVARCHAR:
+                    case Types.LONGNVARCHAR:
+                        VarCharVector varcharVector = (VarCharVector)root.getVector(columnName);
+                        String value = rs.getString(i) != null ? rs.getString(i) : "";
+                        varcharVector.setIndexDefined(rowCount);
+                        varcharVector.setValueLengthSafe(rowCount, value.length());
+                        varcharVector.setSafe(rowCount, value.getBytes(Charset.forName("UTF-8")), 0, value.length());
+                        varcharVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.DATE:
+                        Date date = rs.getDate(i);
+                        DateMilliVector dateMilliVector = (DateMilliVector) root.getVector(columnName);
+                        dateMilliVector.setValueCount(rowCount + 1);
+                        if (date != null) {
+                            dateMilliVector.setSafe(rowCount, rs.getDate(i).getTime());
```

Comment:
There might be a potential issue with timezones. I'm not sure if there's a sane semantic (Java8/JDBC 4.2 has LocalDate...)

Reply:
Yes, you are right. Let me think through on this and get back to you.

Reply:
At this time, I am inclined to just have user provide another parameter such as Calendar so that we can handle the Timezone specific nuances. Many databases don't really consider and handle timezones, daylight savings time and rule changes correctly. Also, it's upto the user to insert the correct values based on the timezone or time related rules. So, what is your opinion on having one additional parameter such as Calendar from end caller?